### PR TITLE
[DBX-61907] Add post-retry-exhaustion hook to 526 backup

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- Leverage Sidekiqs `sidekiq_retries_exhausted` method to track backup path 526 exhaustion.
- Add `rspec-sidekiq` gem to allow testing of Sidekiq lifecycle hooks

## Related issue(s)
- [Ticket to bolster our logging](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/61907)

## Testing done

- Manual running of job to ensure continuity

## What areas of the site does it impact?
526 backup submission monitoring

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/sidekiq/form526_backup_submission_process/submit.rb [hunk 1]
```diff
@@ -21,13 +22,21 @@ class Submit
...
-        Processor.new(form526_submission_id).process!
-        job_status.update(status: Form526JobStatus::STATUS[:success])
+        submission = Form526Submission.find(form526_submission_id)
+        with_tracking('Form526 Backup Submission', submission.saved_claim_id, submission.id) do
+          Processor.new(form526_submission_id).process!
+          job_status.update(status: Form526JobStatus::STATUS[:success])
+        end
       rescue => e
         ::Rails.logger.error(
           message: "FORM526 BACKUP SUMBISSION FAILURE. Investigate immedietly: #{e.message}.",
```
Comment (sethdarragile6) [inline lib/sidekiq/form526_backup_submission_process/submit.rb:42]: I admire your restraint in NOT fixing these typos 🤣

**Review by sethdarragile6 — DISMISSED** [review]
LGTM

**Review by aurora-a-k-a-lightning — APPROVED** [review]

File: lib/sidekiq/form526_backup_submission_process/submit.rb (hunk 1 above)
Comment (SamStuckey) [reply lib/sidekiq/form526_backup_submission_process/submit.rb:42]: oh i just didn't notice lol

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]
rspec-sidekiq gem looks active and legit. And added to the section recommended in [the readme](https://github.com/wspurgin/rspec-sidekiq#installation).

**Pull request merged**